### PR TITLE
OVERHAUL PT. 6: ASSAULT RIFLES REMADE

### PR DESCRIPTION
**Pull request by mazzinsanity**
## Summary:
- Full assault rifle resprite.
- New marksman carbine and service rifle sprites.
- New sprites for 5.56 and 5mm bullet casings
- New sprites for .223 ammo boxes, 7.62 ammo box sprites slightly altered
- Removal of 4.73mm.
- 5mm is now more prevalent.
- Rewritten assault rifle names and descriptions. Some other guns brought up to standard.
- R93 removed, replaced by the Infantry Rifle, a 5mm counterpart to the service rifle.
- Infiltrator is now a unique spawn with buffed stats to reflect its new loot tier.
- Type 93 is now 5mm and burst-fire and in rotation as an R91 counterpart.
- FAL is back in rotation as superhigh.
- G11 is back in rotation as superhigh.
- G11 is an accurate, burst-fire 5mm marksman rifle.
- Bozar now has bonus projectile speed to behave more like a scoped LMG for excellent ranged suppression.
- Chinese LMG is now slower, but has the full AP of 5mm.
- New bullet stats for rifle calibers.
- Additional sprite fixes to previously-added sprites.
- New firing sounds for a number of assault rifles.
- Updated inhand sprites for certain weapons.
- Vexillarius Fox loadout now gets the Type 93 instead of the AK-112.
- **COMPLETELY** changed the way ammo spawners work. You are now going to find boxes and magazines with the spawner tiers reflecting the ammo lathe recipe tiers.
- Restructured weapon loot tiers and weapons no longer appear in several loot tiers at the same time.
- Made an LMG template, repathed LMGs to be its children.
- Fixed an oversight allowing the heavy machine gun to be used to clear out bunkers with ease.

## Image of the new gun sprites:
image

## Ammolathe changes:
- 22lr ammo and mags moved to Tier 1.
- 20 round 5.56 mags and 10 round 7.62mm magazines moved to Tier 2.
- En-bloc clips and sniper magazines moved to Tier 2.
- 5mm boxes and 30 round magazines moved to Tier 3 and require titanium.
- G11 magazines moved to Tier 3.

## New rifle caliber stat details:

5.56 FMJ: Medium damage, Medium wound bonus,
- 25 damage
- 20 wound bonus

.223 FMJ: Less damage, More wound bonus, More damage against simple mobs
- 20 damage
- 30 wound bonus

7.62: High damage, High wound bonus
- 35 damage
- 30 wound bonus

.308 FMJ: Less damage, More wound bonus, More damage against simple mobs
- 30 damage
- 40 wound bonus

5mm FMJ: Low-medium damage, Low-medium wound bonus, High AP
- 20 damage
- 15 wound bonus
- 40% AP

## New pistol caliber stat details:
- 10mm armor penetration removed
- 10mm wound bonus 24 -> 25 (for future ease of balance and consistency by using multiples of 5)
- 9mm damage 16 -> 15 (for future ease of balance and consistency by using multiples of 5)

## New assault rifle stat details:

Selecting semi-auto mode now reduces spread and recoil by ~20% and increases the fire delay.

Assault Rifle (R91) - 5.56mm, 10/20/30/50 Rounds, High tier
- fire_delay = 4
- autofire_shot_delay = 3
- slowdown = 0.4
- spread = 8

Infiltrator (R91 Unique) - 5.56mm, Suppressed, Medium Scope, AP, Accurate, 10/20/30/50 Rounds, Unique tier
- fire_delay = 4 
- autofire_shot_delay = 3
- slowdown = 0.4
- spread = 4
- extra_penetration = 0.3  (30% AP)

Cheap Assault Rifle (AK-112) - 5mm, 30/50 Rounds, High-mid tier
- fire_delay = 4 
- autofire_shot_delay = 3
- slowdown = 0.4
- spread = 16
- extra_daamge = -5  (15 Damage)

Chinese Assault Rifle (Type 93) - 5mm, Burst, AP, 30/50 Rounds, High tier
- burst_size = 3
- fire_delay = 5
- slowdown = 0.4
- spread = 12 

Advanced Assault Rifle (G11) - 5mm, Caseless, Long Scope, AP, Accurate, 50 Rounds, Superhigh tier
- burst_size = 3
- fire_delay = 5
- slowdown = 0.8
- spread = 3
- extra_speed = 700

Automatic Battle Rifle (FAL) - 7.62mm, 10/20 Rounds, Superhigh tier
- fire_delay = 4 
- autofire_shot_delay = 3
- slowdown = 0.5
- spread = 10

## New LMG stat details:

Chinese Light Machine Gun (Type 88) - 5mm, 50 Rounds, High Tier
- autofire_shot_delay = 2
- slowdown = 0.9
- spread = 18

Machine Sniper Rifle (Bozar) - 5.56mm, Long Scope, Extra Projectile Speed, Accurate, 10/20/30/50 Rounds, Superhigh tier
- autofire_shot_delay = 2
- slowdown = 1.1
- spread = 8
- extra_speed = 500

Colt Montior (Scoped BAR) - 7.62mm, AP, Extra Projectile Speed, Medium Scope, Accurate, 10/20 Rounds, Heavy Gunner Exclusive
- autofire_shot_delay = 3
- slowdown = 1.35
- spread = 6
- recoil = 0.6
- extra_damage = -5  (30 Damage)
- extra_penetration = 0.2
- extra_speed = 200

## SMG balance pass:

American 180 slowdown increased: 0.25 -> 0.4
Calico slowdown increased: 0.3 -> 0.4
P90 slowdown increased: 0.2 -> 0.3
P90 spread increased: 13 -> 16
14mm SMG damage decreased: 32 -> 25
14mm SMG armor penetration increased: 0 -> 20
MP5 now deals 20 damage before falloff
Uzi damage malus removed

### CODE REVIEW

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 1]
```diff
@@ -624,375 +641,257 @@
...
-	spread = 2
-	can_unsuppress = FALSE
-	suppressed = 1
-	fire_sound = 'sound/weapons/Gunshot_large_silenced.ogg'
+	casing_ejector = FALSE
+
+//FAL		Keywords: 7.62mm, 10/20 Rounds, Superhigh tier
+/obj/item/gun/ballistic/automatic/assault_rifle/fal
+	name = "automatic battle rifle"
+	desc = "A fully-automatic FAL PARA battle rifle produced by FN Herstal. Intended for use by paratroopers and special forces. Chambered in 7.62mm."
```
Comment (Draganfrukts) [inline code/modules/projectiles/guns/ballistic/automatic.dm:783]:
```suggestion
	desc = "An uniquely powerful Pre-War FAL 50.63 battle rifle developed by FN Herstal. Fitted with a shorter barrel and a lightweight folding stock, intended to be used by paratroopers across the European Commonwealth. Chambered in 7.62mm."
```

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 2]
```diff
@@ -624,375 +641,257 @@
...
-	autofire_shot_delay = 1.7
-	spread = 10
-	recoil = 0.85
-	actions_types = list(/datum/action/item_action/toggle_firemode)
-	fire_sound = 'sound/f13weapons/automaticrifle_BAR.ogg'
-	extra_penetration = 0.25
+//R84 Light Machine Gun			Keywords: 5.56mm, 60 Rounds, High Tier
+/obj/item/gun/ballistic/automatic/lmg/r84
+	name = "light machine gun"
+	desc = "A fully-automatic R84 light machine gun manufactured by FN Herstal. Intended for suppressive fire. Chambered in 5.56mm."
```
Comment (Draganfrukts) [inline code/modules/projectiles/guns/ballistic/automatic.dm:819]:
```suggestion
	desc = "A Pre-War R84 light machine gun designed by FN Herstal and now manufactured by the NCR, commonly issued to Heavy Troopers. Intended for suppressive fire. Chambered in 5.56mm."
```

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 3]
```diff
@@ -624,375 +641,257 @@
...
-	desc = "An especially modified light machine gun built on the BAR with an added pistol grip and a Cutts recoil compensator. This one features a West-Tek marksman scope and its receiver bears a red star."
-	slowdown = 1.35
-	autofire_shot_delay = 2.45
-	spread = 5
-	recoil = 0.2
-	extra_speed = 100
+//Bozar		Keywords: 5.56mm, Long Scope, Extra Projectile Speed, Accurate, 10/20/30/50 Rounds, Superhigh tier
+/obj/item/gun/ballistic/automatic/lmg/bozar
+	name = "\improper Bozar"
+	desc = "A fully-automatic XM67 light machine gun manufactured in small numbers by Heckler & Koch. The ultimate refinement of the sniper's art. Chambered in 5.56mm."
```
Comment (Draganfrukts) [inline code/modules/projectiles/guns/ballistic/automatic.dm:828]:
```suggestion
	desc = "A bullpup Walther XM67 machine sniper rifle that has been designed from the ground up for non-stationary shooting, fitted with a collapsible stock, long-range scope, and reciprocating barrel. The ultimate refinement of the sniper's art. Chambered in 5.56mm."
```

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 4]
```diff
@@ -624,375 +641,257 @@
...
 
-//CHINESE LMG
-/obj/item/gun/ballistic/automatic/rpd
+//Type 88 Machine Gun		Keywords: 5mm, 50 Rounds, High Tier
+/obj/item/gun/ballistic/automatic/lmg/type88
 	name = "chinese light machine gun"
-	desc = "A restored variant of the chinese assault rifle re-chambered to 5mm that feeds from drums, featuring a heavy barrel and a full stock to go with its improved internal mechanism."
-	icon = 'icons/obj/guns/gunfruits2022/lmg.dmi'
-	icon_state = "rpd"
+	desc = "A fully-automatic Type 88 light machine gun manufactured by a Chinese industrial conglomerate. Sports a robust operating mechanism and reinforced stock. Chambered in 5mm."
```
Comment (Draganfrukts) [inline code/modules/projectiles/guns/ballistic/automatic.dm:844]:
```suggestion
	desc = "A Type 88 light machine gun of Chinese design, based on the original success of the Type 93 rifle. It sports a robust operating mechanism and a barrel meant to withstand extreme abuse, on top of it's improved frame. Chambered in 5mm."
```

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 5]
```diff
@@ -624,375 +641,257 @@
...
-	can_attachments = FALSE
-	actions_types = null
-	fire_sound = 'sound/f13weapons/assaultrifle_fire.ogg'
-	extra_penetration = -0.15 //For a total of 30%
 
-//M1919 Machinegun				Keywords: LEGION, .308, Automatic, 80 round belt.
-/obj/item/gun/ballistic/automatic/m1919
+//Colt Monitor		Keywords: 7.62mm, AP, Extra Projectile Speed, Medium Scope, Accurate, 10/20 Rounds, Heavy Gunner Exclusive
+/obj/item/gun/ballistic/automatic/lmg/monitor
+	name = "\improper Colt Monitor"
```
Comment (Draganfrukts) [inline code/modules/projectiles/guns/ballistic/automatic.dm:854]:
```suggestion
	name = "heavy battle rifle"
```

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 6]
```diff
@@ -624,375 +641,257 @@
...
+	desc = "An especially modified light machine gun built on the BAR platform with an added pistol grip and a Cutts recoil compensator. This one features a West-Tek marksman scope and its receiver bears a red star."
```
Comment (Draganfrukts) [inline code/modules/projectiles/guns/ballistic/automatic.dm:855]:
```suggestion
	desc = "An especially modified Pre-War machine gun made for the Federal Bureau of Investigation, built on the famous BAR platform with an added pistol grip and a Cutts recoil compensator. This one has a West-Tek optical gunsight mounted to it's receiver for optimal accuracy. Chambered in 7.62mm."
```

**Review by Draganfrukts — CHANGES_REQUESTED** [review]

**Comment by Fishpoker** [comment]
25 DMG for 5.56 🤣🤣

**Comment by mazzinsanity** [comment]
> 25 DMG for 5.56 🤣🤣

It used to be 26 DMG. This change is aimed at improving consistency by using multiples of 5 and making it easier to balance if needed further down the line

**Comment by HissyNays** [comment]
Why the 10mm and 9mm nerf?

"for future ease of balance and consistency by using multiples of 5" I also really don't like this statement, this isn't ease of balance, this is limitation of balance and greatly harms or buffs guns that are meant to fire lots of bullets fast.

Most auto assault rifles having 3 rate of fire is good due to them getting closer to semi-auto guns in terms of viability but ultimately this is a bad change as you effectively made SMG's overall better option. Rather than curbing assault rifles why not buff bolt actions and semi-autos?

MP-5 now seems to be better than 10mm smg which is strange.

Jesus, I was gonna ask why are you buffing chinese LMG but then I look at previous stats and they are mad, this needs even bigger nerf to stay in line with other LMG's, primarily to slowdown.

The colt monitor nerf was definitely not needed, it has such a big slowdown that it's practicly a meme gun. If it at least had actually big magazine and good fire rate it would make sense, but now it's just an even bigger joke. No wonder I often seen heavy troopers leave colt monitors to collect dust.

Chinese LMG,  colt monitor and R84 LMG badly need to get slowdown lowered/increased at least to 1. This will create a much more fair balance.
I don't understand why chinese LMG given exclusively to legion needs to be exceptional compared to other LMG's in terms of movement speed. You shouldn't be able to effectively chase and push people with LMG.

Semi-auto rifles seem to be entirely forgotten and they are in biggest need of buffs. Marksman carbine needs a scope that is viable. Service rifle remains in lowest tier of weapons, scout carbine is just slightly better service rifle which is saved by being scopeable but still remains subpar.

**Comment by mazzinsanity** [comment]
> Semi-auto rifles seem to be entirely forgotten and they are in biggest need of buffs. Marksman carbine needs a scope that is viable. Service rifle remains in lowest tier of weapons, scout carbine is just slightly better service rifle which is saved by being scopeable but still remains subpar and there's more examples if you dig.

vince-mc-mahon-gulp

Can you guess what guns will be overhauled next?

**Comment by mazzinsanity** [comment]
> MP-5 now seems to be better than 10mm smg which is strange.

The MP5 is meant to be a tier above the 10mm SMG. Same with the Uzi.

**Comment by mazzinsanity** [comment]
> I don't understand why chinese LMG given exclusively to legion...

The Chinese LMG and its blueprint can now be found as High tier bunker loot. Same with the R84.

**Review by Stutternov — APPROVED** [review]

**Comment by HissyNays** [comment]
> > I don't understand why chinese LMG given exclusively to legion...
> 
> The Chinese LMG and its blueprint can now be found as High tier bunker loot. Same with the R84.

Even so, there's still a large gap in slowdown between chinese lmg and other other lmg's which makes little sense. I can Imagine it still stays good for run'n'gunning but we can just wait and see how it is in game meanwhile with other LMG's you might aswell lay down and crawl instead, you won't lose much speed and force everyone to click your sprite directly, this is especially true to heavy trooper role due to SPA.

**Pull request merged**